common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -309,41 +319,84 @@ protected boolean removeTask(Runnable task) {
      * @return {@code true} if and only if at least one task was run
      */
     protected boolean runAllTasks() {
+        assert inEventLoop();
         boolean fetchedAll;
+        boolean ranAtLeastOne = false;
+
         do {
             fetchedAll = fetchFromScheduledTaskQueue();
-            Runnable task = pollTask();
-            if (task == null) {
-                return false;
+            if (runAllTasksFrom(taskQueue)) {
+                ranAtLeastOne = true;
             }
+        } while (!fetchedAll); // keep on processing until we fetched all scheduled tasks.
 
-            for (;;) {
-                try {
-                    task.run();
-                } catch (Throwable t) {
-                    logger.warn("A task raised an exception.", t);
-                }
+        if (ranAtLeastOne) {
+            lastExecutionTime = ScheduledFutureTask.nanoTime();
+        }
+        afterRunningAllTasks(false);
+        return ranAtLeastOne;
+    }
 
-                task = pollTask();
-                if (task == null) {
-                    break;
-                }
+    /**
+     * Runs all tasks from the passed {@code taskQueue}.
+     *
+     * @param taskQueue To poll and execute all tasks.
+     *
+     * @return {@code true} if atleast one task was executed.
+     */
+    protected boolean runAllTasksFrom(Queue<Runnable> taskQueue) {
+        boolean ranAtLeastOne = false;
+        for (;;) {
+            Runnable task = pollTaskFrom(taskQueue);
+            if (task == null) {
+                break;
             }
-        } while (!fetchedAll); // keep on processing until we fetched all scheduled tasks.
-
-        lastExecutionTime = ScheduledFutureTask.nanoTime();
-        return true;
+            ranAtLeastOne = true;
+            try {
+                task.run();
+            } catch (Throwable t) {
+                logger.warn("A task raised an exception.", t);
+            }
+        }
+        return ranAtLeastOne;
     }
 
     /**
      * Poll all tasks from the task queue and run them via {@link Runnable#run()} method.  This method stops running
      * the tasks in the task queue and returns if it ran longer than {@code timeoutNanos}.
      */
     protected boolean runAllTasks(long timeoutNanos) {
+        assert inEventLoop();
         fetchFromScheduledTaskQueue();
-        Runnable task = pollTask();
-        if (task == null) {
+
+        switch (runAllTasksFrom(taskQueue, timeoutNanos, setLastExecutionTime)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)